test/jdk/java/awt/Focus/CrossFocusRequestTest/CrossFocusRequestTest.java

```diff
@@ -0,0 +1,149 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Button;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.TextField;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8307325
+ * @summary Verify the focus owner when non focused windows requesting focus.
+ * @run main CrossFocusRequestTest
+ */
+public class CrossFocusRequestTest {
+
+    private static Frame frame1, frame2;
+    private volatile static Button button;
+    private volatile static TextField textField;
+    private static final int waitTimeout = 1000;
+    private static final CountDownLatch clickLatch = new CountDownLatch(1);
+    private static final CountDownLatch focusLatch = new CountDownLatch(1);
+    private volatile static Point compAt;
+    private volatile static Dimension compSize;
+
+    public static void main(String[] args)
+        throws InvocationTargetException, InterruptedException, AWTException {
+
+        try {
+            EventQueue.invokeAndWait(CrossFocusRequestTest::initializeGUI);
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(() -> {
+                compAt = button.getLocationOnScreen();
+                compSize = button.getSize();
+            });
+
+            robot.mouseMove(compAt.x + compSize.width / 2,
+                compAt.y + compSize.height / 2);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            robot.delay(waitTimeout);
+            if (!clickLatch.await(waitTimeout, TimeUnit.MILLISECONDS)) {
+                throw new RuntimeException(
+                    "FAIL: Button did not trigger actionPerformed when clicked."
+                        + " Test cannot proceed!");
+            }
+
+            robot.delay(waitTimeout);
+
+            EventQueue.invokeAndWait(() -> {
+                compAt = frame1.getLocationOnScreen();
+                compSize = frame1.getSize();
+            });
+
+            robot.mouseMove(compAt.x + compSize.width / 2,
+                compAt.y + compSize.height - 20);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
```

Comment:
How does it test that the `requestFocus` works across frames?!

You click the text field in the first frame, which grants the focus to the text field irrespective whether `requestFocus` was called or not.

It does not make sense to me.